### ISSUE TITLE
Visual Studio 2017 RC support

### ISSUE THREAD

**Issue by Narvalex** [Microsoft/VS-PPT#66]
I think the title is self explanatory

**Comment by justcla** [comment]
Got it. We're onto it. ;-)

**Comment by grantbowering** [comment]
Thank goodness it's on its way... Horizontal tabs is so jarring now!  :smile:

**Comment by albertgouws** [comment]
Without sounding like a nag, do you have some sort of ETA? (is it days, weeks, months?)

My favorite features I can't live without are

- Vertical tabs
- Matching selected text in the right hand margin (This is so much faster and nicer than doing a find!)
- Fix Mixed Tabs

Thanks!

**Comment by justcla** [comment]
It is days. Hopefully I'll have this up early December.
(I was planning to do more work on it today.)

**Comment by justcla** [comment]
Actually, we've already started working on it. Just in the process of updating some references to use NuGet.
Once that's done, it's just a matter of package and release!
(But there are a LOT of references to updates.)

**Comment by verysimplenick** [comment]
Any progress? :)

**Comment by justcla** [comment]
Yeah, we've moved all the references to NuGet. But are having some conflicts with some dependencies that might not be backward compatible. We're just working through that.
We're getting there. ;-)

**Comment by justcla** [comment]
Update: We have a built version (unsigned) that runs in VS2017.
We just need to test it and get it signed, then we can release it.

If you'd like to help with the testing, let me know and I'll get you a copy of the pre-release.

**Comment by chrfin** [comment]
@justcla I would really like to help testing!

**Comment by verysimplenick** [comment]
@justcla yep, I want help with testing too VS ver. 15.0.25914.0

**Comment by chrfin** [comment]
@verysimplenick Are you missing the RC update? I'm on 15.0.26009.0...

**Comment by verysimplenick** [comment]
@chrfin yes, not installed yet. If need for PPT tests latest RC - can update :)

**Comment by badjester** [comment]
@justcla Hey. I would like to help testing, too. I'm already using VS 2017 as default on our current projects.

**Comment by gius** [comment]
I can also help testing if more users are needed.

**Comment by luetm** [comment]
I'd like to help with testing!

**Comment by AlcatrazDev** [comment]
I'd like to help testing as well!

**Comment by AltEnter** [comment]
I'd like to help with testing！！！

**Comment by AndersMad** [comment]
I'd love to get a copy too! will you provide a dl link? i haven't started using 2017 yet because it us "useless" with only a few horizontal tabs - need them vertical 80'ish colored tabs 😄

**Comment by luisfco** [comment]
@justcla I can test too, using VS2017 as default now.

**Comment by AlcatrazDev** [comment]
How could we get the copy of the pre-release for testing?

**Comment by ebalders** [comment]
@justcla I would like to test he new version.

**Comment by DoCode** [comment]
@justcla, I am too ;)
+1

**Comment by justcla** [comment]
An update for everyone....
I'm waiting to release the code for testing, but going through the process of getting everything digitally signed. That means getting it to run on an internal build server, and I'm having some issues with some of the project references and the new v15.0 build tools.
Sorry for the delay. Hopefully we'll have something out in the next day or so.
I will be posting it to the [releases ](https://github.com/Microsoft/VS-PPT/releases) section of this GitHub site.

**Comment by DoCode** [comment]
@justcla, always having issues?

**Comment by schuettecarsten** [comment]
@justcla If you need testing, let me know :)

**Comment by AlcatrazDev** [comment]
@justcla Any news? we really need this extension!

**Comment by justcla** [comment]
Sorry for the delay. Christmas/New Year break slows things down.

Update:
We are right now considering a new strategy for releasing PPT.
We are considering a model that would install each of the features/components as individual extensions, which would allow you to only install/enable just the components you want, and manage them individually through Extensions and Updates.
So we're looking at options for an installer that will let you choose which components are installed
We're also considering utilizing Virtual Private Galleries to deliver an updated/curated list of experimental features. (Would appear in E&U dialog as an entry under "Online" and "Updates")

If you've got any feedback about these options, please share your thoughts.

**Comment by schuettecarsten** [comment]
@justcla, maybe you should talk to Mads Kristensen how he solves this for his WebExtensionPack extension: https://marketplace.visualstudio.com/items?itemName=MadsKristensen.WebExtensionPack

**Comment by justcla** [comment]
@schuettecarsten Yes. That's exactly what we're looking at replicating. ;-)
We think that will be a good solution for downloading and installing the components that you want.
It currently downloads and installs all that are not currently installed. It might be good to modify this to let you confirm which ones get downloaded and installed.

Also, we can drop in a Virtual Private Gallery pgkdef so that you can browse the component list via the Extensions and Updates dialog (so you can easily install/uninstall other components without running the installer again).

**Comment by ebalders** [comment]
@justcla While those sound like great features, it would be nice to get the current feature set released with VS2017 support. 

Can PPT be released as it currently exists, with work on individual extensions continuing afterward?

**Comment by Frankwin** [comment]
Any more news on this? Would love my Power Tools back. Just upgraded to VS 2017 RC and I miss them very much.

**Comment by tualatin** [comment]
The same here, I need PPT to VS2017 ;) Please make a pre release... :)

**Comment by creage** [comment]
I miss MatchMargin in VS2017 very badly!

**Comment by AlcatrazDev** [comment]
@justcla We would love to use the extensions, we have been waiting for them a lot of time, more than a month now.

We need them, if you are unsure how to provide them, then just create a pre-release or something, seems like you're enjoying making us waiting.

**Comment by PhilBarila** [comment]
What a bunch of previous comments said.  Cleaning it up and making it nice is great, but I'll take it less cleaned up and nice if I can have it now.   Deliver customer value early = good.  Deliver perfect later, not as much.

**Comment by luetm** [comment]
@justcla I think people in this thread should calm down and let you guys do your job. I love the extension and I'm looking forward to it - when it's done. If people are hellbent on using a RC they must accept the fact that some plugins will not be available for a while. Please remember that probably most of us feel this way. ;)

Have a nice week!

**Comment by justcla** [comment]
Thanks @luetm 
And for everyone, I appreciate your patience.
I'm hoping to upload something today.
I've broken the extensions into 15 separate extensions, which will be individually downloadable from the Marketplace.
Standby. ;-)

**Comment by AndersMad** [comment]
wow.. the last few comments and the fact that @luetm comment got downvotes = o.. m.. f.. g!... thanks @justcla for providing this add-on (!) on a RC (!!) baring the refactoring in mind despise the above.. sry for the spam subscribers :)

**Comment by PhilBarila** [comment]
@justcla, thank you for the great tools.  The fact that we're passionate about them says much for their usefulness.
As one who was down-voted for that passion, I wanted to say that I was really only trying to say that we (software devs in general) should keep our eyes on delivering customer value early, and not get caught by the allure of "Just a little bit better here ..."
I've certainly been seduced by that siren song.

**Comment by justcla** [comment]
It's worth noting that no matter how quickly users want things released, there are a bunch of things that absolutely must be done before any kind of release - including making sure that all components are signed, that appropriate licenses are in place and approved, that the product names and ID are defined (as we don't want to change these in the next version), and many other tasks. Microsoft are not a going to release unsigned, non-licensed components, despite vocal user demands.
These tasks take time, they involve many people, and we've just come out of Christmas/New Year break, where much of the office (including me!) is enjoying downtime.

We also want to make sure that the solution we put out will work for now, for later, and well into the future. A poorly considered, rushed release can set in stone many problems that we (as a community) will struggle with if not handled properly in the first instance.

So, thanks for your patience while we work through these very appropriate steps.
The release is imminently pending.

**Comment by AlcatrazDev** [comment]
@justcla It's more about the fact that you said a month ago that you were going to release a pre-release for testing and that it was almost ready.

![image](http://image.prntscr.com/image/d9beb74146fa48c0947b7153ddef95d4.png)

If it was not ready then don't say anything, I would have just prefered "We are working on it".

Anyway, I would like to thank everybody working on this extension, and all the efforts you're doing to make it working on a RC!

**Comment by DoCode** [comment]
@AlcatrazDev, @justcla I think the same. The message a month ago is misleading. People have a wrong expectation when you say that the release is available shortly. So please communicate a real time slot for the release date. Better to say early that the release is available in 3 Month.

**Comment by justcla** [comment]
The wait is over. I've published 15 new extensions to the Marketplace.
Feel free to download them and try them on your instances of VS2017 RC.
Enjoy!

https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.AlignAssignments
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.CopyAsHtml
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.CtrlClickGoToDefinition
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.CustomDocumentWell
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.Double-ClickMaximize
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.EditorGuidelines
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.FixMixedTabs
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.MatchMargin
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.MiddleClickScroll
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.PeekHelp
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.QuickLaunchTasks
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.SolutionErrorVisualizer
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.PowerCommandsforVisualStudio
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.SyntacticLineCompression
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.TimeStampMargin

**Comment by justcla** [comment]
Note, we are planning to release an installer that will download and install all PPT extensions that are not already installed.
(No promises about when this will come out!)

**Comment by justcla** [comment]
One of the easiest ways to batch install extensions is to do it from inside the VS2017 Extensions and Updates dialog.
Search for "Productivity" and you can select each of the extensions you want to install.
When you close VS, it will install all extensions in one go.

**Comment by Frankwin** [comment]
I tried to install "Ctrl+Click Go To Definition" and I got an error. Here's the install.log

`1/18/2017 10:06:50 AM - Microsoft VSIX Installer
1/18/2017 10:06:50 AM - -------------------------------------------
1/18/2017 10:06:50 AM - vsixinstaller.exe version:
1/18/2017 10:06:50 AM - 15.0.26020.0 built by: D15REL
1/18/2017 10:06:50 AM - -------------------------------------------
1/18/2017 10:06:50 AM - Command line parameters:
1/18/2017 10:06:50 AM - C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VSIXInstaller.exe,C:\Users\frank\Downloads\CtrlClickGoToDef (1).vsix
1/18/2017 10:06:50 AM - -------------------------------------------
1/18/2017 10:06:50 AM - Microsoft VSIX Installer
1/18/2017 10:06:50 AM - -------------------------------------------
1/18/2017 10:06:50 AM - Initializing Install...
1/18/2017 10:06:51 AM - Extension Details...
1/18/2017 10:06:51 AM - 	Identifier         : CtrlClickGoToDefn.b66d9c08-a6c6-4fbf-9827-ba4693968419
1/18/2017 10:06:51 AM - 	Name               : Ctrl+Click Go To Definition
1/18/2017 10:06:51 AM - 	Author             : Microsoft
1/18/2017 10:06:51 AM - 	Version            : 2.0
1/18/2017 10:06:51 AM - 	Description        : Make ctrl+click perform a "Go To Definition" on the identifier under the cursor.
Also, when the ctrl key is held down, highlight identifiers that look like they have definitions to navigate to.
1/18/2017 10:06:51 AM - 	Locale             : en-US
1/18/2017 10:06:51 AM - 	MoreInfoURL        : http://blogs.msdn.com/noahric
1/18/2017 10:06:51 AM - 	InstalledByMSI     : False
1/18/2017 10:06:51 AM - 	SupportedFrameworkVersionRange : [4.5,)
1/18/2017 10:06:51 AM - 
1/18/2017 10:06:51 AM - 	SignatureState     : ValidSignature
1/18/2017 10:06:51 AM - 	SignedBy           : Microsoft Corporation
1/18/2017 10:06:51 AM - 	Certificate Info   : 
1/18/2017 10:06:51 AM - 		-------------------------------------------------------
1/18/2017 10:06:51 AM - 		[Subject]       : CN=Microsoft Corporation, OU=MOPR, OU=OPC, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
1/18/2017 10:06:51 AM - 		[Issuer]        : CN=Microsoft Code Signing PCA 2010, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
1/18/2017 10:06:51 AM - 		[Serial Number] : 33000000FF6B54F91DD1781A1C0000000000FF
1/18/2017 10:06:51 AM - 		[Not Before]    : 2/11/2016 6:12:29 PM
1/18/2017 10:06:51 AM - 		[Not After]     : 5/11/2017 7:12:29 PM
1/18/2017 10:06:51 AM - 		[Thumbprint]    : 6F1691367DB1D2F6278B8A82A1C4095D3BC0AF1C
1/18/2017 10:06:51 AM - 
1/18/2017 10:06:51 AM - 	Supported Products : 
1/18/2017 10:06:51 AM - 		Microsoft.VisualStudio.Community
1/18/2017 10:06:51 AM - 			Version : [15.0]
1/18/2017 10:06:51 AM - 
1/18/2017 10:06:51 AM - 	References         : 
1/18/2017 10:06:51 AM - 		-------------------------------------------------------
1/18/2017 10:06:51 AM - 		Identifier   : PPTOptionsPage.666715a6-3ac4-4bb7-b538-4d7625f99666
1/18/2017 10:06:51 AM - 		Name         : Productivity Power Tools Options Page
1/18/2017 10:06:51 AM - 		Version      : [2.0,3.0)
1/18/2017 10:06:51 AM - 		MoreInfoURL  : 
1/18/2017 10:06:51 AM - 		Nested       : Yes
1/18/2017 10:06:51 AM - 
1/18/2017 10:06:51 AM - 	Prerequisites      : 
1/18/2017 10:06:51 AM - 		-------------------------------------------------------
1/18/2017 10:06:51 AM - 		Identifier   : Microsoft.VisualStudio.Component.CoreEditor
1/18/2017 10:06:51 AM - 		Name         : Visual Studio core editor
1/18/2017 10:06:51 AM - 		Version      : [15.0.25904.2,16.0)
1/18/2017 10:06:51 AM - 
1/18/2017 10:06:51 AM - Signature Details...
1/18/2017 10:06:51 AM - 	Extension is signed with a valid signature.
1/18/2017 10:06:51 AM - 
1/18/2017 10:06:51 AM - Searching for applicable products...
1/18/2017 10:06:51 AM - Found installed product - Global Location
1/18/2017 10:06:51 AM - Found installed product - Visual Studio Community 2017 RC
1/18/2017 10:06:52 AM - The extension with ID 'CtrlClickGoToDefn.b66d9c08-a6c6-4fbf-9827-ba4693968419' is not installed to Visual Studio Community 2017 RC.
1/18/2017 10:06:55 AM - The following target products have been selected...
1/18/2017 10:06:55 AM - 	Visual Studio Community 2017 RC
1/18/2017 10:06:55 AM - 
1/18/2017 10:06:56 AM - Beginning to install extension to Visual Studio Community 2017 RC...
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64 declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64 doesn't declare a chip
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:56 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.Extensions.X64 declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.Extensions.X64 doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64 declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64 doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.ConcurrencyCheck.X64.Resources doesn't declare a chip
1/18/2017 10:06:57 AM - Dependency Microsoft.VisualCpp.CodeAnalysis.Extensions.X64 declares Chip=x64, but package Microsoft.VisualCpp.CodeAnalysis.Extensions.X64 doesn't declare a chip
1/18/2017 10:06:57 AM - Install Error : Microsoft.VisualStudio.ExtensionManager.NestedExtensionInstallException: The extension 'PPTOptionsPage.666715a6-3ac4-4bb7-b538-4d7625f99666' nested within 'CtrlClickGoToDefn.b66d9c08-a6c6-4fbf-9827-ba4693968419' failed to install.
Object reference not set to an instance of an object. ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.VisualStudio.ExtensionManager.InstallableExtensionImpl.RefreshExtractedPackageStream(IInstalledExtension parentExtension, IExtensionReference reference)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.InstallNestedExtensions(InstallableExtensionImpl parentExtension, IInstalledExtension installedParentExtension, ZipPackage parentVSIX, InstallFlags installFlags, IDictionary`2 extensionsInstalledSoFar, List`1 extensionsUninstalledSoFar, IInstalledExtensionList modifiedInstalledExtensionsList, AsyncOperation asyncOp, IProgress`1 progress)
   --- End of inner exception stack trace ---
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.InstallNestedExtensions(InstallableExtensionImpl parentExtension, IInstalledExtension installedParentExtension, ZipPackage parentVSIX, InstallFlags installFlags, IDictionary`2 extensionsInstalledSoFar, List`1 extensionsUninstalledSoFar, IInstalledExtensionList modifiedInstalledExtensionsList, AsyncOperation asyncOp, IProgress`1 progress)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.PerformLegacyInstall(InstallableExtensionImpl extension, IDictionary`2 extensionsInstalledSoFar, List`1 extensionsUninstalledSoFar, IInstalledExtensionList modifiedInstalledExtensionsList, IProgress`1 progress, InstallFlags installFlags, AsyncOperation asyncOp, Boolean isPackComponent, IInstalledExtension& newExtension)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.PerformSetupEngineInstall(InstallableExtensionImpl extension, Boolean installPerMachine, Boolean isPackComponent, IDictionary`2 extensionsInstalledSoFar, List`1 extensionsUninstalledSoFar, IInstalledExtensionList modifiedInstalledExtensionsList, IProgress`1 progress, InstallFlags installFlags, AsyncOperation asyncOp, IInstalledExtension& newExtension)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.InstallInternal(InstallableExtensionImpl extension, InstallFlags installFlags, IDictionary`2 extensionsInstalledSoFar, List`1 extensionsUninstalledSoFar, IInstalledExtensionList modifiedInstalledExtensionsList, AsyncOperation asyncOp, IProgress`1 progress)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.BeginInstall(IInstallableExtension installableExtension, InstallFlags installFlags, AsyncOperation asyncOp)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.InstallWorker(IInstallableExtension extension, InstallFlags installFlags, AsyncOperation asyncOp)
`

**Comment by justcla** [comment]
@Frankwin Can you please start a separate issue for that error you're getting?
Thanks.

**Comment by justcla** [comment]
All - please raise separate issues for any bugs you find with any of the new extensions.
Thanks.

**Comment by AlcatrazDev** [comment]
Thank you, any idea if the extension StructureVisualizer will be released?

**Comment by olegtk** [comment]
@AlcatrazDev, Structure Visualizer is now part of Visual Studio, a bit trimmed down version feature-wise though. Are you missing any of its features that didn't make it into Visual Studio?

**Comment by AlcatrazDev** [comment]
@olegtk that's not comparable

![image](http://image.prntscr.com/image/9dfb9d1de0494957b92325bd15a814f8.png)

**Comment by olegtk** [comment]
@AlcatrazDev you mean you don't like the dashed pattern of the indent lines?

**Comment by AlcatrazDev** [comment]
@olegtk if I ask for the Structure Visualizer extension... it's most likely because I don't like the dashed pattern, don't you think? :smile_cat: 

I like the Structure visualizer colors and the indent lines.

**Comment by olegtk** [comment]
@AlcatrazDev I see. Please file a bug via "Help/Send Feedback" about dashed lines. Overall I'm asking because we cut some Structure Visualizer features when porting it into Visual Studio (scroll bar map mode integration for example) and I'm looking for feedback about that.

**Comment by justcla** [comment]
@AlcatrazDev There's no need to be rude when our engineers are kind enough to take time to listen and respond to your comments. If you can't be constructive, your comments will be removed by the moderators.

If you ask for Structure Visualizer, and we've already included it in the product, we're going to be wondering what it is that you think you're missing.

> if I ask for the Structure Visualizer extension... it's most likely because I don't like the dashed pattern, don't you think?

No - I wouldn't think so at all. There is no implied link between "Where is feature X?" and "I would like a slight modification to feature X."

When we implemented Structure Visualizer in the product, we made slight changes to the lines (now using dashes, not using colors), we improved the info box that appears when you hover over the lines, and we removed the component that showed structure on the vertical margin.
We are interested in feedback about this. In particular, whether users are missing the margin feature.
We will take as feedback that you prefer solid lines to dashed lines.

**Comment by justcla** [comment]
Early testing feedback:

There are four extensions that install the PPTOptionsPage along with the extension.
- Copy As Html
- Match Margin
- Syntactic Line Compression
- Time Stamp Margin

It seems some users are experiencing problems installing these four components.
It is likely related to the included extension PPTOptionsPage.

**Comment by AlcatrazDev** [comment]
@olegtk I didn't know that the current Structure visualizer in the RC was this one but trimmed down, I apologize for my last comment it was out of context.

I did not read your question properly, I thought you asked if I don't like the dashed pattern or the indent lines, I thought you were refering to the left picture with indent lines, my bad.

@justcla My intention was not to be rude, I apologize if it seemed so

About the Structure visualizer, an option to switch between old structure visualizer look and the dashed pattern would be nice.

**Comment by olegtk** [comment]
@AlcatrazDev we probably haven't documented it correctly, but yes, Structure Visualizer was ported into Visual Studio 2017 core editor. Btw, props to our intern @jepetty for doing most of the work! 

Thanks for feedback about dashed lines, but if you want it to get it fixed earlier, please submit it yourself via Help/Send Feedback or http://visualstudio.uservoice.com (formal feedback from external users gets way more attention).

**Comment by creage** [comment]
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.MatchMargin this one is not available at MarketPlace :(

**Comment by chrfin** [comment]
@creage Please see the following issue: https://github.com/Microsoft/VS-PPT/issues/77 -> not available until RC3

**Comment by justcla** [comment]
I have temporarily disabled the following extensions due to the bug listed here:
[Nested Extensions not working pre RC3](https://github.com/Microsoft/VS-PPT/issues/77)
- Ctrl+Click GoTo Definition
- Match Margin
- Syntactic Line Compression
- Time Stamp Margin

**Comment by schuettecarsten** [comment]
@justcla Do you plan to push the current source code into this repository?

**Comment by justcla** [comment]
Until the fix for Nested Extensions is publicly available in RC3, I have removed the PPTOptionsPage extension from each of the four failing extensions and made it available as a separately downloadable/installable extension.
PPT Options Page: [https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.ProductivityPowerToolsOptionsPage](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.ProductivityPowerToolsOptionsPage)

I have re-published the four extensions:
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.CopyAsHtml
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.MatchMargin
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.SyntacticLineCompression
https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.TimeStampMargin

**Comment by justcla** [comment]
# Productivity Power Tools VS2017

For Visual Studio 2017, the individual components of Productivity Power Tools have been broken into separate extensions.
You can download them here:

* [Align Assignments](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.AlignAssignments)
* [Copy As Html](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.CopyAsHtml)
* [Ctrl+Click GoTo Definition](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.CtrlClickGoToDefinition)
* [Custom Document Well](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.CustomDocumentWell)
* [Double-Click Maximize](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.Double-ClickMaximize)
* [Editor Guidelines](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.EditorGuidelines)
* [Fix Mixed Tabs](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.FixMixedTabs)
* [Match Margin](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.MatchMargin)
* [Middle-Click Scroll](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.MiddleClickScroll)
* [Peek Help](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.PeekHelp)
* [Quick Launch Tasks](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.QuickLaunchTasks)
* [Solution Error Visualizer](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.SolutionErrorVisualizer)
* [Power Commands for Visual Studio](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.PowerCommandsforVisualStudio)
* [Syntactic Line Compression](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.SyntacticLineCompression)
* [Time Stamp Margin](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.TimeStampMargin)
* [PPT Options Page](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.ProductivityPowerToolsOptionsPage)

**Comment by usmanfarooq-pk** [comment]
Awesome work guys. Dash pattern looks more clean and gives nice look. Structure visualizer popup is also very nice. Just one thing... can we have colors back in vertical line ? Because colors tell us that which block we are currently in without doing extra effort.

**Comment by markw-t** [comment]
Thanks for the comment @usmanfarooq-pk . When you think about how you used the colors in the line, were you using them to determine what the blocks meant semantically (loop vs if block vs other structure) or simply to distinguish one block's lines from another? In other words, is the semantic mapping of line color important to you, or simply the fact that the lines have different colors? @hemchans and I would love to hear your feedback here.

**Comment by luisfco** [comment]
@markw-t I also want colored structure visualizer lines please !!
Semantic mapping would be nice (like In CodeRush, right now I have both CR+StructureVisualizer enabled at the same time), but just to have different colors (like in previous version) it's ok.

Nice work !!!!

**Comment by usmanfarooq-pk** [comment]
@markw-t thanks for your reply. I use productivity tools with VS2015. Its structure visualizer vertical lines give us different colors for different code blocks. For example, classes have light blue color. Methods have royal blue color (i guess). Loops have purple color and if-else have green color.

When I am coding in C# editor I know that if cursor is inside a purple color then I am writing my code inside a loop. So I don't need to hover over a vertical line or scroll up to see that which block i am currently coding in.

If you can bring these colors in VS2017 then it will be a BIG help for us. Thanks!

**Comment by markw-t** [comment]
Thanks @usmanfarooq-pk . I've started a UserVoice vote for this at https://visualstudio.uservoice.com/forums/121579-visual-studio-ide/suggestions/18628639-improve-the-structure-visualizer-in-visual-studio - please do go ahead and vote if you are passionate about this issue, as it will help us prioritize.

**Comment by usmanfarooq-pk** [comment]
@markw-t Thanks! I have cast my vote along with comments.

**Comment by justcla** [comment]
@usmanfarooq-pk Thanks for your input.

Just to confirm, you like the fact that the various colors represent specific structure types (ie. loop, if-statement, etc).
So assigning random colors to each line does not satisfy your wish.

How it differs:
Option 1: Colors assigned to structure types.
If you four nested have nested if statements, you'll see four lines of the same color.
Option 2: Colors assigned based on indent level.
If you have four nested is statements, each would have a unique color.
A while loop in one part of the code might have the same colored line as an if-statement in a different part.

**Comment by usmanfarooq-pk** [comment]
In my opinion, **Option 1 would be better.** 

Option 2 seems bit confusing to me. For example if we have 4 nested if-statements then having 4 different colors will confuse us. It wouldn't give us clear picture that which block we are currently coding in. Remember, the idea is to differentiate structure types and Option 1 will serves this purpose very well. However, if someone wants to differentiate 4 nested if-statements then he can use hover over option which will show him little tool tip explain him that where he is :)

**Another little suggestion:**

Vertical line can be highlighted if we hover over on it OR if we select opening/closing brace of code block. This option would be very helpful if we have looooong code blocks. By the way its also helpful in short code blocks. But for long code blocks this feature will really help a lot. I guess Notepad++ has this option.

**Comment by justcla** [comment]
Fantastic feedback. Thanks @usmanfarooq-pk 
All good suggestions. Will definitely consider this for future implementations.
